flink/src/main/java/org/apache/iceberg/flink/sink/IcebergFilesCommitter.java

```diff
@@ -85,9 +88,9 @@
   // iceberg table when the next checkpoint happen.
   private final NavigableMap<Long, byte[]> dataFilesPerCheckpoint = Maps.newTreeMap();
 
-  // The data files cache for current checkpoint. Once the snapshot barrier received, it will be flushed to the
+  // The completed files cache for current checkpoint. Once the snapshot barrier received, it will be flushed to the
   // 'dataFilesPerCheckpoint'.
-  private final List<DataFile> dataFilesOfCurrentCheckpoint = Lists.newArrayList();
+  private final List<WriteResult> writeResultsOfCurrentCkpt = Lists.newArrayList();
```

Comment:
Is it correct for this to be a list of write results if a write result keeps track of a list of data files and a list of delete files?

Reply:
Yes, it's correct here. Because if there're 5 `IcebergStreamWriter`, then each writer will emit a `WriteResult`.  For the one parallelism `IcebergFilesCommitter`,  it will collect all the `WriteResult`(s) in this `writeResultsOfCurrentCkpt` cache,  and then merge them into a single `WriteResult`.  Finally,  write those files into delete + data manifests and update the flink statebackend.